storage/credential.go

```diff
@@ -153,7 +153,7 @@ func (cred *AADSASCredential) getToken(ctx context.Context) (SASToken, error) {
 	defer span.End()
 
 	if cred.token != nil {
-		if cred.token.expiry.Before(time.Now().Add(-5 * time.Minute)) {
+		if !cred.token.expiry.Before(time.Now().Add(-5 * time.Minute)) {
```

Comment:
I am a little confused about the logic here. After your modification, when the expiry time is one minute ago, it will still directly return the token? Maybe the negative sign needs to be removed.

Reply:
I was thinking the same. I am removing negative time. With this change now a new token will be generated every 55 minutes. Which is fine.